### PR TITLE
expose listener to manually update wayname

### PR DESCRIPTION
**Pull request by tobrun**
Extracts SDK code for https://github.com/mapbox/mapbox-navigation-android/pull/1344

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 1]
```diff
@@ -565,8 +566,20 @@ private void removeAllMarkers() {
...
+    if (mapWayname.isAutoQueryEnabled() && onUpdateWaynameListener != null) {
+      onUpdateWaynameListener.updateWaynameWithLocation(location);
+    } else {
+      LatLng latLng = new LatLng(location);
+      PointF mapPoint = mapboxMap.getProjection().toScreenLocation(latLng);
+      mapWayname.updateWaynameWithPoint(mapPoint, waynameLayer);
+    }
+  }
+
+  public void setOnUpdateWaynameListener(OnUpdateWaynameListener onUpdateWayname) {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Could you add Javadocs here?

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Also, could you place this `public` method with the rest (before `private` ones)?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 2]
```diff
@@ -565,8 +566,20 @@ private void removeAllMarkers() {
...
+    this.onUpdateWaynameListener = onUpdateWayname;
+  }
+
+  public interface OnUpdateWaynameListener {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Could you extract `OnUpdateWaynameListener` from `NavigationMapboxMap` into a separate file?

**Comment by tobrun** [comment]
Thank you both for 👀! Considering that another approach will be taken, I'm going to close this PR.

**Pull request closed without merging**